src/masternodes/mn_checks.cpp

```diff
@@ -1504,3 +1513,139 @@ bool IsMempooledCustomTxCreate(const CTxMemPool & pool, const uint256 & txid)
+        return Res::Err("%s: %s", __func__, "malformed tx vouts (wrong number of vouts)");
+    }
+
+    CFulfillOrderImplemetation fillorder;
+    CDataStream ss(metadata, SER_NETWORK, PROTOCOL_VERSION);
+    ss >> static_cast<CFulfillOrder &>(fillorder);
+    if (!ss.empty()) {
+        return Res::Err("%s: deserialization failed: excess %d bytes", __func__, ss.size());
+    }
+    
+    fillorder.creationTx=tx.GetHash();
+    fillorder.creationHeight = height;
+
+    CTxDestination ownerDest = DecodeDestination(fillorder.ownerAddress);
+    if (ownerDest.which() == 0) {
+        return Res::Err("%s: %s", __func__, "ownerAdress (" + fillorder.ownerAddress + ") does not refer to any valid address");
+    }
+    if (!mnview.GetOrderByCreationTx(fillorder.orderTx)) {
+        return Res::Err("order with creation tx %s does not exists!", fillorder.orderTx.GetHex());
+    }
+
+    // Return here to avoid already exist error
+    if (rpcInfo) {
+        rpcInfo->pushKV("ownerAddress", fillorder.ownerAddress);
+        rpcInfo->pushKV("orderTx", fillorder.orderTx.GetHex());
+        rpcInfo->pushKV("amount", fillorder.amount);
+        return Res::Ok();
+    }
+
+    auto res = mnview.FulfillOrder(fillorder);
+    if (!res.ok) {
+        return Res::Err("%s %s: %s", __func__, fillorder.creationTx.GetHex(), res.msg);
+    }
+
+    return Res::Ok();
+}
+
+Res ApplyCloseOrderTx(CCustomCSView & mnview, CCoinsViewCache const & coins, CTransaction const & tx, uint32_t height, std::vector<unsigned char> const & metadata, Consensus::Params const & consensusParams, bool skipAuth, UniValue *rpcInfo)
+{
+    // Check quick conditions first
+    if (tx.vout.size() !=2) {
+        return Res::Err("%s: %s", __func__, "malformed tx vouts (wrong number of vouts)");
+    }
+
+    CCloseOrderImplemetation closeorder;
+    CDataStream ss(metadata, SER_NETWORK, PROTOCOL_VERSION);
+    ss >> static_cast<CCloseOrder &>(closeorder);
+    if (!ss.empty()) {
+        return Res::Err("%s: deserialization failed: excess %d bytes", __func__, ss.size());
+    }
+    
+    closeorder.creationTx=tx.GetHash();
+    closeorder.creationHeight = height;
+
+    std::unique_ptr<COrderImplemetation> order;
+    if (!(order=mnview.GetOrderByCreationTx(closeorder.orderTx))) {
+        return Res::Err("order with creation tx %s does not exists!", closeorder.orderTx.GetHex());
+    }
+
+    order->closeTx=closeorder.creationTx;
+    order->closeHeight=closeorder.creationHeight;
+
+    // Return here to avoid already exist error
+    if (rpcInfo) {
+        rpcInfo->pushKV("orderTx", closeorder.orderTx.GetHex());
+        return Res::Ok();
+    }
+
+    // auto res = mnview.CloseOrder(closeorder);
+    // if (!res.ok) {
+    //     return Res::Err("%s %s: %s", __func__, closeorder.creationTx.GetHex(), res.msg);
+    // }
+    // res = mnview.CloseOrderTx(*order);
+    // if (!res.ok) {
+    //     return Res::Err("%s %s: %s", __func__, closeorder.creationTx.GetHex(), res.msg);
+    // }
```

Comment:
This function does nothing ?

Reply:
I have uncommented that part yesterday in some fix.